apps/server/tests/factories.py

```
@@ -0,0 +1,54 @@
+"""Shared test-data factories for the craftcontrol backend test suite.
+
+Use these builders instead of constructing raw dicts inline so that envelope
+and snapshot shapes stay consistent when the schema evolves.
+"""
+from __future__ import annotations
+
+
+def telemetry_envelope(
+    type: str = "blocks.changed",
```

Ruff (0.16.3)

[error] Function argument `type` is shadowing a Python builtin

(A002)